apis/goharbor.io/v1alpha2/chartmuseum_types.go

```diff
@@ -0,0 +1,356 @@
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+	AccessSecretRef string `json:"accessSecretRef,omitempty"`
+}
+
+type ChartMuseumChartStorageDriverOpenStackSpec struct {
+	// +kubebuilder:validation:Required
+	// Container to store charts for openstack storage backend
+	Container string `json:"container"`
+
+	// +kubebuilder:validation:Optional
+	// Prefix to store charts for the container
+	Prefix string `json:"prefix,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Region of the container
+	Region string `json:"region,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// URL for obtaining an auth token.
+	// https://storage.myprovider.com/v2.0 or https://storage.myprovider.com/v3/auth
+	AuthenticationURL string `json:"authenticationURL"`
+
+	// +kubebuilder:validation:Optional
+	// Your Openstack tenant name.
+	// You can either use tenant or tenantid.
+	Tenant string `json:"tenant,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Your Openstack tenant ID.
+	// You can either use tenant or tenantid.
+	TenantID string `json:"tenantID,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Your Openstack domain name for Identity v3 API. You can either use domain or domainid.
+	Domain string `json:"domain,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Your Openstack domain ID for Identity v3 API. You can either use domain or domainid.
+	DomainID string `json:"domainID,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// The Openstack user name. You can either use username or userid.
+	Username string `json:"username,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// The Openstack user id. You can either use username or userid.
+	UserID string `json:"userid,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*"
+	// Secret name containing the Openstack password.
+	PasswordRef string `json:"passwordRef,omitempty"`
+}
+
+type ChartMuseumChartStorageDriverFilesystemSpec struct {
+	// +kubebuilder:validation:Required
+	VolumeSource corev1.VolumeSource `json:"volumeSource"`
+
+	// +kubebuilder:validation:Optionel
+	Prefix string `json:"prefix,omitempty"`
+}
+
+type ChartMuseumChartIndexSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=0
+	// Parallel scan limit for the repo indexer
+	ParallelLimit *int32 `json:"parallelLimit,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Pattern="([0-9]+h)?([0-9]+m)?([0-9]+s)?([0-9]+ms)?([0-9]+us)?([0-9]+µs)?([0-9]+ns)?"
+	// Timestamp drift tolerated between cached and generated index before invalidation
+	StorageTimestampTolerance *metav1.Duration `json:"storageTimestampTolerance,omitempty"`
+}
+
+type ChartMuseumPostFormFieldNameSpec struct {
```

Comment:
Simiar case to form fields. They have fixed values.

```
CHART_POST_FORM_FIELD_NAME=chart
PROV_POST_FORM_FIELD_NAME=prov
```

Reply:
I suggest t add comments to warn about that.
Following discussion on <https://github.com/goharbor/harbor-operator/pull/59#discussion_r467071357>